Classes/YTPlayerView.m

```diff
@@ -831,7 +852,31 @@ - (NSString *)stringFromVideoIdArray:(NSArray *)videoIds {
  * @return JavaScript response from evaluating code.
  */
 - (NSString *)stringFromEvaluatingJavaScript:(NSString *)jsToExecute {
-  return [self.webView stringByEvaluatingJavaScriptFromString:jsToExecute];
+    dispatch_semaphore_t sema4 = dispatch_semaphore_create(0);
+    __block NSString *resultString = nil;
+
+    [self.webView evaluateJavaScript:jsToExecute completionHandler:^(id result, NSError *error) {
+        if (error == nil && result != nil) {
+            resultString = [NSString stringWithFormat:@"%@", result];
+        }
+        dispatch_semaphore_signal(sema4);
+    }];
+    // WkWebView always calls completion handler on the main thread, so this keeps us from blocking ourselves.
+    // c.f. http://stackoverflow.com/questions/28388197/wkwebview-trying-to-query-javascript-synchronously-from-the-main-thread
+    while (dispatch_semaphore_wait(sema4, DISPATCH_TIME_NOW)) {
+        [[NSRunLoop currentRunLoop] runMode:NSDefaultRunLoopMode
+                                 beforeDate:[NSDate dateWithTimeIntervalSinceNow:10]];
+    }
```

Comment:
why 10 seconds? isn't 500ms or one second enough? also, is there a reason we must wait for the completion before allowing the flow to continue?